src/indexer/index.ts

```diff
@@ -82,6 +82,17 @@ const contracts: { [contract in Contract]: ContractInfo } = {
   },
 }
 
+async function getBlockTimestamps(events: EventLog[], kit: ContractKit) {
+  const blockNumberToTimestamp: Record<number, number> = {}
+  await Promise.all(
+    events.map(async ({ blockNumber }) => {
+      const block = await kit.web3.eth.getBlock(blockNumber)
```

Comment:
are you sure all events will have a different block? if not, seems inefficient to do it this way

Reply:
Good point, code now only calls `getBlock` for the set of unique block numbers